go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/testcontainers/testcontainers-go
 
-go 1.22
+go 1.23.0
+
+toolchain go1.23.6
```

Comment:
question: with this specific toolchain directive in all the go modules, will our GHA build the project with Go 1.24 to validate it's working in that latest major, or will Go downgrade the toolchain to the defined here?